lib/nostrum/struct/component.ex

```diff
@@ -75,13 +75,27 @@ defmodule Nostrum.Struct.Component do
   defmacro __using__(_opts) do
     quote do
       alias Nostrum.Struct.Component
-      alias Nostrum.Struct.Component.{ActionRow, Button, SelectMenu, SelectMenuOption}
       alias Nostrum.Struct.Emoji
+      alias Nostrum.Struct.Component.{Button, SelectMenu, SelectMenuOption}
       alias Nostrum.Util
+
+      def new(opts \\ []) do
+        opts
+        |> Enum.into(@defaults)
+        |> to_component()
+      end
+
+      def to_component(map), do: Component.to_struct(map)
     end
   end
 
+  @doc """
+  Each component type implements this callback to be provided with a keyword list which will produce a probably-valid component of that type.
```

Comment:
This "probably-valid" does not quite spark trust to users of the library. Does it produce valid components, or does it not? When could it not produce valid components? If we can hand out the reasons for this not generating valid components, that would already be a good thing to do.

Reply:
Good point.

I was alluding to the fact that a user of the library _could_ attempt to create a component of `type: 2, label: "hello", placeholder: "🧀"` these are all valid component fields and they are typed correctly. But it is not a valid button.

That is what I was referring to here:

> These would greatly benefit from some helper functions that I showed when I initially asked for comment. 

The way I imagined it could be user-proofed was to include functions such as `create_link_button(label, url, emoji)` to ensure that components produced are always valid in regards to the combination of fields as well the contents of the fields themselves.

Reply:
Currently, there's only the predefined set of components by Discord though, right?

If not, then we should probably document that Nostrum does not validate component types defined outside of the library. That way, users know which parts to use and which not to use.

Reply:
Yes. Currently just the button, action row, and select.